src/metrics/metric_manager.py

```diff
@@ -0,0 +1,230 @@
+            for phase in self.phases:
+                if phase not in phase_mapping:
+                    raise KeyError(f'Phase has no key = {phase}, it must be one of {list(phase_mapping.keys())}')
+                else:
+                    phases.append(phase_mapping[phase])
+            self.phases = phases
+
+
+class MetricWithUtils(nn.Module):
+    """Union class for metric and metric utils parameters.
+    
+    Args:
+        metric: Metric written with TorchMetrics.
+        mapping: Dictionary for mapping Metric forward input keys with Task output dictionary keys.
+        log_name: The metric name used in logs.
+    """
+    def __init__(self, metric: Metric, mapping: Dict[str, str], log_name: str):
+        super().__init__()
+        self.__metric = metric
+        self.__mapping = mapping
+        self.__log_name = log_name
+
+    @property
+    def log_name(self) -> str:
+        """The metric name used in logs."""
+        return self.__log_name
+
+    @property
+    def mapping(self) -> Dict[str, str]:
+        """Dictionary for mapping Metric forward input keys with Task output dictionary keys."""
+        return self.__mapping
+
+    def forward(self, *args, **kwargs):
+        return self.__metric(*args, **kwargs)
+
+    def compute(self):
+        return self.__metric.compute()
+
+
+class MetricManager(nn.Module):
+    """Manages all metrics for a Task.
+    
+    Args:
+        params: Metric parameters.
+    """
+    # model use phases
+    phases = [Phase.TRAIN, Phase.VALID, Phase.TEST, Phase.PREDICT]
+
+    def __init__(self, params: List[MetricParams]):
+        super().__init__()
+        # Change list to set in phases
+        for param in params:
+            param.phases = set(param.phases)
+
+        self.__phase2metrics = nn.ModuleDict()
+        for phase in Phase:
+            self.__phase2metrics[phase.name] = self.__get_phase_metrics(params, phase)
+
+    def __get_phase_metrics(self, params: List[MetricParams], phase: Phase) -> nn.ModuleList:
+        """Generate metric list for current phase.
+
+        Args:
+            params: All metric params from config file.
+            phase: Current phase Enum.
+
+        Returns:
+            metrics: Metric list as nn.ModuleList for current phase. 
+        """
+        # create added_metric_names list
+        added_log_names = []
+        metrics = []
+        for metric_params in params:
+            if phase not in metric_params.phases:
+                continue
+            metric = METRICS.get(metric_params.name)(**metric_params.params)
+            mapping = metric_params.mapping
+            prefix = '' if metric_params.prefix is None else metric_params.prefix + '_'
+            log_name = prefix + metric_params.name
+            if log_name in added_log_names:
+                # If prefix not set.
+                if metric_params.prefix is None:
+                    raise ValueError('Has a two identical metrics. Please, set in config file '
```

Comment:
"a" is not used in front of plural nouns in English. Also, I think you don't need defferent behaviours depending on prefix being set. Just use one:
`Got two metrics with identical names: {log_name}. Please, set differet prefixes for identical metrics in the config file`